### DISCUSSION TITLE
CSL boundaries rotation axis and angle

### DISCUSSION THREAD

**Discussion by HPouraliakbar** [mtex-toolbox/mtex#2156] [Ask Anything] [answered]
Hi folks,

I would like to know which rotation axis and angle are considered when we use the CSL command in MTEX to calculate CSL boundaries. For example, CSL13 can have multiple rotation combinations. How can we differentiate between these different rotations, or how can we select a specific one? What are those default rotations with commands?

**Comment by kilir** [comment, marked as answer]
Hi,
Mtex returns all misorientations related to a given sigma,
e.g.
```Matlab
cs = crystalSymmetry('cubic')
csl13 = CSL(13,cs)


% csl13 = misorientation (m-3m → m-3m)
% size: 1 x 2
 
%  Bunge Euler angles in degree
%     phi1     Phi    phi2
%  53.1301 22.6199  323.13
%  22.6199       0       0

```
If you want to pick one you can do so by `csl13(2)` (with respect to above variable names). 
If you want to see how it is computed, have  a look into the function `CSL`.

Hope that helps.

Cheers,

Rüdiger

**Reply by HPouraliakbar** [reply]
Hi Rüdiger,

Thanks for the reply and apologies for my late reply.

I have a couple of questions that I wish you could help me with them:

I would like to get the rotation axis and rotation angle for CSL boundaries. For example for CSL27, I get this info:

csl27 = misorientation (m-3m → m-3m)
  size: 1 x 2
 
  Bunge Euler angles in degree
     phi1     Phi    phi2
  105.945 15.6425 285.945
   101.31 22.1916  281.31

1) is there any specific plane(s) or direction(s) that must be parallel for CSL27? I assume as there is no info, the only important parameters are the rotation angles (Phi1,Phi,Phi2) as mentioned above

2) How can I plot those two types of rotations separately? I cannot introduce CSL27(2) or CSL27(1). I get errors when I run the command.

Thanks,
Hesam

**Comment by HPouraliakbar** [comment]
Another thing that I highly need to know is why we put a threshold in the commands.

gB3 = gB(angle(gB.misorientation,CSL(3,ebsd.CS)) < 5*degree);

for instance, in the above line, the threshold of 5 is considered. How do we decide about the threshold?

**Reply by kilir** [reply]
Hi,

>I would like to get the rotation axis and rotation angle for CSL boundaries. For example for CSL27, I get this info:

In order to get the axis/angle representation

```
axis(csl27)
angle(csl27)
```
should do.


>How can I plot those two types of rotations separately? I cannot introduce CSL27(2) or CSL27(1). I get errors when I run the command.

That remains a little unclear to me. What error do you get?

Cheers,

Rüdiger

**Reply by kilir** [reply]
> `gB3 = gB(angle(gB.misorientation,CSL(3,ebsd.CS)) < 5*degree);`
>for instance, in the above line, the threshold of 5 is considered. How do we decide about the threshold?

This depends on your actual structure. There's Brandons criterion and derivations of it  e.g. https://doi.org/10.1007/s10853-006-0665-8

Cheers,

Rüdiger

**Reply by HPouraliakbar** [reply]
regarding the axis/angle, the command does not show the info I need. Please note I am using MTEX 5.10.2:

>> axis csl3
Error using axis (line 236)
Unknown command option csl3.
 
>> angle csl3

ans =

     0     0     0     0

**Reply by HPouraliakbar** [reply]
Regarding the higher order CSLs (higher than 11), I cannot plot them and I get an error. I think this can be related to different possibilities of rotations which need to be defined first, for instance, by asking CSL13(1) or CSL13(2) we should define the exact rotation.

For example, I use the following command for CSL11:
gB11 = gB(angle(gB.misorientation,CSL(11,ebsd.CS)) < 3*degree);

but, I cannot use the same format for CSL13. This the error I get:

gB13 = gB(angle(gB.misorientation,CSL(13,ebsd.CS)) < 3*degree);
Error using bsxfun
Non-singleton dimensions of the two input arrays must match each other.

Error in orientation/dot (line 166)
  d = abs(bsxfun(@times,o1.a,o2.a) + bsxfun(@times,o1.b,o2.b) + ...

Error in orientation/angle (line 23)
  omega = real(2*acos(abs(dot(o1,varargin{:}))));

Also:
csl13(2)=CSL(13(2),cs)
 csl13(2)=CSL(13(2),cs)
                ↑
Error: Invalid expression. When calling a function or indexing a variable, use parentheses.
Otherwise, check for mismatched delimiters.
 
Did you mean:
>> csl13(2)=CSL(13*(2),cs)

**Reply by kilir** [reply]
Hi,
for the syntax of e.g. function axis, see here:  https://mtex-toolbox.github.io/orientation.axis.html

I do not get any errors for that example:

```
mtexdata martensite
grains = ebsd.calcGrains
%%
gB = grains.boundary('i','i')
gB13 = gB(angle(gB.misorientation,CSL(13,ebsd.CS)) < 3*degree);
```
using mtex 5.12, Matlab R2023b.

Cheers,

Rüdiger

**Reply by HPouraliakbar** [reply]
I tried exactly your codes for CSL13 and got the following error:

gB13 = gB(angle(gB.misorientation,CSL(13,ebsd.CS)) < 3*degree);

%%
Error using bsxfun
Non-singleton dimensions of the two input arrays must match each other.

Error in orientation/dot (line 166)
  d = abs(bsxfun(@times,o1.a,o2.a) + bsxfun(@times,o1.b,o2.b) + ...

Error in orientation/angle (line 23)
  omega = real(2*acos(abs(dot(o1,varargin{:}))));
%%

Could you please help me how to define two different CSL13 rotations?
